### PR TITLE
198 Ability to lock user in HSL ID

### PR DESCRIPTION
**Pull request by Divinare**
Closes https://github.com/HSLdevcom/bultti/issues/198

* Steps for thesting can be found from the issue description.

* when user is logged and then blocked from HSL ID, he can still do things like saving (until he presses f5)
   * there is nothing we can do about this (or we could fetch groups from HSL ID with every request but that's not what we want to do)
   * BUT if you make user.role = 'BLOCKED' by hand in DB, then saving should be prevented -> test that this works (works for me at least)
 
* when user is blocked, it gives 400 (Bad Request) and that seems to crash the app. Maybe not in scope of this task to fix it but maybe it should be fixed somewhere (or here)?

* Note: from now on, user must be 'blocked' in HSL ID (if making it only 'blocked' in DB, it will be overridden when the user logins if he is not in 'blocked group')

### CODE REVIEW

File: src/util/useAuth.ts [hunk 1]
```diff
@@ -18,17 +19,17 @@ export enum AuthState {
 export const useAuth = (): [AuthState, boolean] => {
   const [authState, setAuthState] = useState<AuthState>(AuthState.UNAUTHENTICATED)
   const [currentUser, setCurrentUser] = useStateValue('user')
+  let [, setErrorMessage] = useStateValue('errorMessage')
   // To prevent unwanted navigation, only set this to true when the app should
   // navigate away from the login screen.
   let shouldNavigate = useRef(false)
-
-  const [login, { loading: loginLoading }] = useMutationData<User>(loginMutation)
+  const [login, { loading: isLogingLoading }] = useMutationData<User>(loginMutation)
```
Comment (danieldunderfelt) [inline src/util/useAuth.ts:26]: Typo?

File: src/util/useAuth.ts [hunk 2]
```diff
@@ -115,10 +120,11 @@ export const useAuth = (): [AuthState, boolean] => {
           refetchUser()
         } else {
           setAuthState(AuthState.UNAUTHENTICATED)
+          setIsLoginFailed(true)
```
Comment (danieldunderfelt) [inline src/util/useAuth.ts:123]:
```suggestion
          setErrorMessage(text('login_failed'))
```
Can't the message be set here? A new state variable for this seems excessive.

**Review by danieldunderfelt — CHANGES_REQUESTED** [review]
A few comments, otherwise good!

File: src/util/useAuth.ts (hunk 2 above)
Comment (Divinare) [reply src/util/useAuth.ts:123]: Hmm true! This code changed a bit on the way and didn't think setting it here after setIsLoginFailed variable was made!

**Comment by Divinare** [comment]
Review fixes done

File: src/util/useAuth.ts [hunk 3]
```diff
@@ -18,17 +19,16 @@ export enum AuthState {
 export const useAuth = (): [AuthState, boolean] => {
   const [authState, setAuthState] = useState<AuthState>(AuthState.UNAUTHENTICATED)
   const [currentUser, setCurrentUser] = useStateValue('user')
+  let [, setErrorMessage] = useStateValue('errorMessage')
   // To prevent unwanted navigation, only set this to true when the app should
   // navigate away from the login screen.
   let shouldNavigate = useRef(false)
-
-  const [login, { loading: loginLoading }] = useMutationData<User>(loginMutation)
+  const [login, { loading: isLoggingLoading }] = useMutationData<User>(loginMutation)
```
Comment (danieldunderfelt) [inline src/util/useAuth.ts:26]:
```suggestion
  const [login, { loading: isLoginLoading }] = useMutationData<User>(loginMutation)
```

File: src/util/useAuth.ts [hunk 4]
```diff
@@ -115,10 +115,11 @@ export const useAuth = (): [AuthState, boolean] => {
...
         } else {
           setAuthState(AuthState.UNAUTHENTICATED)
+          setErrorMessage(text('login_failed'))
         }
       })
     }
   }, [currentUser, codeUrlParam, authState, login, refetchUser, navigateNext])
 
-  return [authState, loginLoading]
+  return [authState, isLoggingLoading]
```
Comment (danieldunderfelt) [inline src/util/useAuth.ts:124]:
```suggestion
  return [authState, isLoginLoading]
```

**Review by danieldunderfelt — APPROVED** [review]
LGTM! The typo fix went in a bit of a wrong direction, but I'll just merge the suggested changes.

**Pull request merged**